### PR TITLE
Cleanup/http request

### PR DESCRIPTION
**Pull request by bassosimone**

### CODE REVIEW

**Comment by bassosimone** [comment]
There is a segmentation fault in regress tests using g++. Currently investigating this (it is always nice that those kind of things pop up with one compiler and not with another :-)

**Comment by AntonioLangiu** [comment]
I think it'is important to have a build matrix on Travis that checks such things for us.

**Comment by bassosimone** [comment]
In general a build matrix is useful and I am totally in favor of it. But, in this case, it was enough to have g++ and clang++ to spot the issue. My comment was more of regret because I tested this on my Mac using clang++ and hence I did not spotted the segfault and opened the pull request.

**Comment by bassosimone** [comment]
I've rebased and restarted the build. Interestingly it fails again (good that this thing is quite deterministic) and also interestingly cannot reproduce this bug using Vagrant and 14.04.

Will download 12.04 with Vagrant and try again next.

**Comment by bassosimone** [comment]
#FastFact: issue is gone when switching to 14.04. Since the issue only appeared for g++ (and so, I imagine, GNU's libc++), I'm starting to wonder whether this had to do with a too young GNU C++ implementation in Ubuntu 12.04. It may have sense to apply b7b13ab775 to other recent failing pull requests (i.e. #398 and #396) and see whether this fixes them as well (maybe I'd run more than one test in a row to increase confidence that everything is good).

File: include/measurement_kit/http.hpp [hunk 1]
```diff
@@ -131,6 +131,45 @@ Url parse_url(std::string url);
...
+/// \param body Optional HTTP request body.
+/// \param lp Optional logger.
+/// \param pol Optional poller.
+/// \param client Optional HTTP client.
+inline void GET(std::string url, RequestCallback cb,
+                Headers headers = {}, std::string body = "",
+                Settings settings = {}, Logger *lp = Logger::global(),
+                Poller *pol = Poller::global(),
+                Var<Client> client = Client::global()) {
+    settings["method"] = "GET";
```
Comment (hellais) [inline include/measurement_kit/http.hpp]: I think that the `method` in the settings should not be overriden if it has been specified/passed.

Perhaps a check to ensure if it's set and if not set it by default to get is enough.

Comment (hellais) [inline include/measurement_kit/http.hpp]: Ah no, nevermind I wasn't reading the class definition properly. This is ok. Ignore my previous comment.

**Comment by hellais** [comment]
I have tested this and I like the new syntactic sugar. I am going to proceed in merging this.

**Pull request merged**